### PR TITLE
Remove deprecated functionality

### PR DESCRIPTION
**Pull request by alexcrichton**
This removes a large array of deprecated functionality, regardless of how
recently it was deprecated. The purpose of this commit is to clean out the
standard libraries and compiler for the upcoming alpha release.

Some notable compiler changes were to enable warnings for all now-deprecated
command line arguments (previously the deprecated versions were silently
accepted) as well as removing deriving(Zero) entirely (the trait was removed).

The distribution no longer contains the libtime or libregex_macros crates. Both
of these have been deprecated for some time and are available externally.

### CODE REVIEW

File: src/libcollections/dlist.rs [hunk 1]
```diff
@@ -864,88 +771,6 @@ mod tests {
     }
 
     #[test]
-    #[allow(deprecated)]
-    fn test_append() {
```
Comment (aturon) [inline src/libcollections/dlist.rs]: append is still with us

Comment (aturon) [inline src/libcollections/dlist.rs]: ... for now

Comment (Gankro) [inline src/libcollections/dlist.rs]: :fearful:

**Comment by Gankro** [comment]
:confetti_ball:

File: src/libcollections/slice.rs [hunk 2]
```diff
@@ -2127,22 +2032,6 @@ mod tests {
     }
 
     #[test]
-    fn test_partition() {
```
Comment (aturon) [inline src/libcollections/slice.rs]: can you change these to test the new partition function in iter (possibly moving them if you think there's a better spot)? i neglected to add tests for it...

File: src/libcollections/str.rs [hunk 3]
```diff
@@ -407,215 +372,6 @@ macro_rules! utf8_acc_cont_byte {
...
 
-/*
-Section: MaybeOwned
-*/
-
-/// A string type that can hold either a `String` or a `&str`.
-/// This can be useful as an optimization when an allocation is sometimes
-/// needed but not always.
-#[deprecated = "use std::string::CowString"]
-pub enum MaybeOwned<'a> {
```
Comment (aturon) [inline src/libcollections/str.rs]: \o/  killed it dead

**Comment by aturon** [comment]
soooo clean! r=me after a rebase

File: src/libcollections/slice.rs (hunk 2 above)
Comment (alexcrichton) [inline src/libcollections/slice.rs]: These tests were actually duplicated in `vec.rs` for some reasons, but I kept those!

File: src/libcollections/dlist.rs (hunk 1 above)
Comment (alexcrichton) [inline src/libcollections/dlist.rs]: oops!

**Pull request merged**